src/SQLite.Net/TableMapping.cs

```diff
@@ -184,51 +235,76 @@ public Column(PropertyInfo prop, CreateFlags createFlags = CreateFlags.None, ICo
                 {
                     var typeCol = propType.GetTypeInfo().GenericTypeArguments;
-                    if (typeCol.Length > 0)
+                    var nullableType = typeCol[0];
+                    var baseType = nullableType.GetTypeInfo().BaseType;
+                    if (baseType == typeof(Enum))
+                    {
+                        valueToSet = AsEnumValue(obj, nullableType, val);
+                    }
+                    else
                     {
-                        var nullableType = typeCol[0];
-                        var baseType = nullableType.GetTypeInfo().BaseType;
-                        if (baseType == typeof (Enum))
-                        {
-                            SetEnumValue(obj, nullableType, val);
-                        }
-                        else
-                        {
-                            _prop.SetValue(obj, val, null);
-                        }
+                        valueToSet = val;
                     }
                 }
-                else if (typeInfo.BaseType == typeof (Enum))
+                else if (typeInfo.BaseType == typeof(Enum))
                 {
-                    SetEnumValue(obj, propType, val);
+                    valueToSet = AsEnumValue(obj, propType, val);
                 }
                 else
                 {
-                    _prop.SetValue(obj, val, null);
+                    valueToSet = val;
                 }
-            }
 
-            private void SetEnumValue(object obj, Type type, object value)
-            {
-                var result = value;
-                if (result != null)
+                // If we're a value tuple then we need to recreate the tuple with the new value and set that
+                // on the property.
+                if (TupleElement != -1)
+                {
+                    var tupleObj = (ITuple)_infoProvider.GetValue(_prop, obj);
+                    var args = new object[tupleObj.Length];
+                    for (var i = 0; i < tupleObj.Length; ++i)
+                    {
+                        args[i] = i == TupleElement ? valueToSet : tupleObj[i];
+                    }
+
+                    valueToSet = Activator.CreateInstance(_infoProvider.GetMemberType(_prop), args);
+                }
+
+                switch (_prop)
                 {
-                    result = Enum.ToObject(type, result);
-                    _prop.SetValue(obj, result, null);
+                    case PropertyInfo pi:
+                        pi.SetValue(obj, valueToSet);
+                        break;
+                    case FieldInfo fi:
+                        fi.SetValue(obj, valueToSet);
+                        break;
                 }
             }
 
-    
+            private object AsEnumValue(object obj, Type type, object? value)
+            {
+                var result = value ?? 0;
```

Comment:
this seems like new behavior to default to 0?

Reply:
This is a behaviour change, as we now always set the value. Prior to the change the field would not get its value set if the passed in value was null.